pliers/utils/updater.py

```diff
@@ -0,0 +1,80 @@
+""" Utility to check if results have changed in foreign APIs """
+import glob
+import datetime
+import warnings
+import pandas as pd
+import numpy as np
+from os.path import realpath, join, dirname, exists, expanduser
+
+from pliers.stimuli import load_stims
+from pliers.extractors import merge_results
+
+def check_updates(extractors, datastore=None, stimuli=None):
+    """ Run graph_spec on set of stimuli, and store results in datastore csv.
+    If file exists, compare results to previous in file,
+    and return if any features have changed
+    Args:
+        extractors - A list of extractor objets.
+        datastore - filepath of CSV file with results. Stored in home dir
+                    by default
+        stimuli - list of stimuli file paths to extract from.
+                  if None, use test data.
+    """
+    # Check if file exists
+    if datastore is None:
+        datastore = expanduser('~/.pliers_updates')
+    if not exists(datastore):
+        prior_data = None
+    else:
+        prior_data = pd.read_csv(datastore)
+
+    if stimuli is None:
+        stimuli = glob.glob(join(
+            dirname(
+                realpath(__file__)), '../tests/data/image/CC0/*'))
+
+    # Load stimuli and extract new features
+    stimuli = load_stims(stimuli)
+
+    results = []
+    for stim in stimuli:
+        for ext in extractors:
+            results.append(ext.transform(stim))
+
+    ## Merge results
+    results = merge_results(results).drop(
+        ['source_file', 'filename', 'history', 'class', 'onset', 'duration'],
+        axis=1, level=0)
+    results['time_extracted'] = datetime.datetime.now()
+
+    # Flatten columns and make into one row
+    results = results.pivot(columns='stim_name', index='time_extracted')
+    results.columns = ['.'.join(col).strip() \
```

Comment:
I believe that this should effectively be done if `flatten_columns=True` is passed into `merge_results` (will be separated by underscore instead of period. Along the same lines `metadata=False` can also be passed into `merge_results` and then in L46 you would only need to drop `onset` and `duration`.

Reply:
I can't remember why I didn't go without because I tried it, but I'll give it another go. Thanks.